api/product.rb

```diff
@@ -18,22 +18,59 @@
 module Api
   # Repesents a product to be managed
   class Product < Api::Object::Named
-    attr_reader :base_url
     attr_reader :objects
     attr_reader :prefix
     attr_reader :scopes
+    attr_reader :versions
 
     include Compile::Core
 
     def validate
       super
       set_variables @objects, :__product
-      check_property :base_url, String
       check_property :objects, Array
       check_property_list :objects, Api::Resource
       check_property :prefix, String
       check_property :scopes, ::Array
       check_property_list :scopes, String
+
+      check_versions
+    end
+
+    # Represents a version of the API for this product
+    class Version < Api::Object
+      attr_reader :base_url
+      attr_reader :default
+      attr_reader :name
+
+      def validate
+        super
+        check_property :base_url, String
+        check_property :name, String
+        check_optional_property :default, [TrueClass, FalseClass]
+      end
+    end
+
+    def default_version
+      @versions.each do |v|
```

Comment:
Consider adding `return @versions.last if @versions.length == 1` and adjusting the `check_versions` below. I am fine also with requiring default to be set to true even if we only have one version.

Reply:
Done.